### DISCUSSION TITLE
generate a pdf of a route that requires laravel authentication

### DISCUSSION THREAD

**Discussion by elbouziani** [spatie/browsershot#878] [Q&A]
I want to generate a pdf of a route that requires laravel authentication, I want to log the user with the Auth:login() function before starting the generation, does anyone have an idea, thank you

**Comment by FoxRocks76** [comment]
I believe this is what you're looking for:

```->authenticate('username', 'password')```

https://spatie.be/docs/browsershot/v4/miscellaneous-options/using-http-authentication
